charms/worker/k8s/src/charm.py

```diff
@@ -149,10 +151,47 @@ def __init__(self, *args):
         )
 
         self.framework.observe(self.on.update_status, self._on_update_status)
+        self.framework.observe(self.on.feature_relation_changed, self._on_feature_relation_changed)
+
         if self.is_control_plane:
             self.etcd = EtcdReactiveRequires(self)
             self.framework.observe(self.on.get_kubeconfig_action, self._get_external_kubeconfig)
 
+    def _on_feature_relation_changed(self, event: ops.RelationChangedEvent):
```

Comment:
so, once this is called from the `reconciler(...)` -- the signature will change a bit.  We won't know which event we're responding to -- it may not even be a relation event.  What we should check instead is 
* are there any feature relations
* do those feature relations have unit data
* can any of the relation data be interpreted as snap config